src/lightning/pytorch/loops/prediction_loop.py

```diff
@@ -114,17 +119,34 @@ def run(self) -> Optional[_PREDICT_OUTPUT]:
     def setup_data(self) -> None:
         trainer = self.trainer
         source = self._data_source
-        pl_module = trainer.lightning_module
         # a dfault `predict_step` exists in the LightningModule, so no need to check if it's overridden
         if not source.is_defined() or trainer.limit_predict_batches == 0:
             return
 
-        self.max_batches, combined_loader = trainer._data_connector._reset_eval_dataloader(
-            RunningStage.PREDICTING, model=pl_module
-        )
```

Comment:
isn't there a resolve_overfit_batches missing somewhere here? It was part of the _reset_eval_dataloader, and as I understand, the code was partially moved here.

Reply:
You are correct that the overfit batches check was run before and not anymore. However, I don't see any source mentioning that `overfit_batches` is meant to work with `trainer.predict()`. Do you think it should? What would be the benefit?

Reply:
Actually, we were doing it unnecessarily for `trainer.validate/test`: [`9320d34` (#16918)](https://github.com/Lightning-AI/lightning/pull/16918/commits/9320d34cddea954664d803eba80dd44bc2cf96b4)